### ISSUE TITLE
Regression bug in the tab bar of the editor: the mouseover and selected highlights do not show properly

### ISSUE THREAD

**Issue by jnsebgosselin** [spyder-ide/spyder#15327]
## Problem Description

The mouseover and selected highlights in the tabbar of the editor are gone. I think this was introduced in PR #15124.

![tabbar_editor](https://user-images.githubusercontent.com/10170372/115127755-08beaa00-9fa7-11eb-8992-b934a3dc45b3.gif)

## Versions
* Spyder version: 5.1.0.dev0 d005fce99
* Python version: 3.7.6 64-bit
* Qt version: 5.12.5
* PyQt5 version: 5.12.3
* Operating System: Windows 10

## Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  3.0.4 (OK)
cloudpickle >=0.5.0           :  1.5.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.6.0               :  7.16.1 (OK)
jedi =0.17.2                  :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  21.2.1 (OK)
nbconvert >=4.0               :  5.6.1 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.1 (OK)
parso =0.7.0                  :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.7.2 (OK)
pygments >=2.0                :  2.6.1 (OK)
pylint >=1.0                  :  2.5.3 (OK)
pyls >=0.36.2;<1.0.0          :  v5.0.1+1.gd005fce99 (NOK)
pyls_black >=0.4.6            :  0.4.6 (OK)
pyls_spyder >=0.3.2           :  0.3.2 (OK)
qdarkstyle =3.0.2             :  3.0.dev (NOK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.1.0.dev0 (OK)
qtconsole >=5.0.3             :  5.0.3 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.8.3                 :  0.9.4 (OK)
setuptools >=39.0.0           :  49.6.0.post20210108 (OK)
sphinx >=0.6.6                :  3.1.2 (OK)
spyder_kernels >=2.0.1;<2.1.0 :  2.1.0.dev0 (OK)
textdistance >=4.2.0          :  4.2.0 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0      :  0.10.3 (OK)
zmq >=17                      :  19.0.1 (OK)

# Optional:
cython >=0.21                 :  0.29.21 (OK)
matplotlib >=2.0.0            :  3.2.1 (OK)
numpy >=1.7                   :  1.20.2 (OK)
pandas >=1.1.1                :  1.1.4 (OK)
scipy >=0.17.0                :  1.5.3 (OK)
sympy >=0.7.3                 :  None (NOK)
```

**Comment by jnsebgosselin** [comment]
Also, the x icons are not painted correctly.

**Comment by stonebig** [comment]
how strange, I juste recompiled Spyder-5.0.11 source to remove PyQt5-12 constraint, and my defautl coloring is quite different
![image](https://user-images.githubusercontent.com/4312421/115139033-fd639100-a02f-11eb-865e-d45a566cf492.png)

**Comment by jnsebgosselin** [comment]
This is interesting, you are right, it looks ok for you.

I am starting it from the `bootstrap.py` and get this error, maybe this is why?

```
No QCoreApplication instance found. Application patches not applied. You have to call load_stylesheet function after instantiation of QApplication to take effect.
```

**Comment by jnsebgosselin** [comment]
If I checkout at `c6dc102f86c13a1fe89853c293a752419cd3da06` it looks fine.

**Comment by jnsebgosselin** [comment]
OK I was wrong, the problem does not occur after PR #15124 was merged for me, but after PR #15237 instead.

**Comment by jnsebgosselin** [comment]
I'll try to investigate later and see what exactly is causing this.

**Comment by stonebig** [comment]
may do a `pip download --dest  D:\tempo spyder` and see what goes down, to check you didn't miss a dependancy update

**Comment by jnsebgosselin** [comment]
The problem is caused by the stylesheet applied to the `EditorQSplitter`. It seems to override the stylesheet applied to the 

If I comment out the line `self.setStyleSheet(self._stylesheet)` in the `EditorQSplitter`, then everything seems to look great.

![image](https://user-images.githubusercontent.com/10170372/115150065-c93e9f00-a034-11eb-927d-27ed8d3d7ac9.png)

**Comment by stonebig** [comment]
this line ? https://github.com/spyder-ide/spyder/blob/e582a41b95f07f1d27b2fa1a1891ac6dca69475d/spyder/plugins/editor/widgets/editor.py#L2921

**Comment by jnsebgosselin** [comment]
Yes.

**Comment by stonebig** [comment]
I have this line in my working set...
.. but I noticed trying to get my red color, some things are not well working in "Theme" change.
My "Theme" is
![image](https://user-images.githubusercontent.com/4312421/115150619-7e8b4a00-a069-11eb-91e7-16050ce9e1ab.png)

**Comment by ccordoba12** [comment]
@jnsebgosselin, where are we applying an style to EditorStack? I think we are not doing that.

**Comment by jnsebgosselin** [comment]
Yes you are right.

It seems to override the stylesheet that are applied through `APP_STYLESHEET, APP_TOOLBAR_STYLESHEET, PANES_TABBAR_STYLESHEET`, that is what I meant.

**Comment by jnsebgosselin** [comment]
I don't understand why it is doing that for me, but not for stonebig...

**Comment by stonebig** [comment]
I'm using PyQt-5.15.1 , a hacked-for-pyside2 QtPy, and a slow PC.

**Comment by jnsebgosselin** [comment]
Thank you stonebig for you help, it is much appreciated!

Maybe this is because of my version of PyQt. I am using the latest one on conda-forge, which is version 5.12.5.

**Comment by stonebig** [comment]
If you're on Conda, the whole compiler toolchain is different.

**Comment by ccordoba12** [comment]
This could be a problem on Windows only because I can't reproduce it on Linux. I'll check it out.

**Comment by jnsebgosselin** [comment]
@ccordoba12 Thank you!

**Comment by jnsebgosselin** [comment]
It is only the main splitter that is affected. The ones that are created when actually splitting the editor looks great.

![image](https://user-images.githubusercontent.com/10170372/115152758-78cd3e80-a040-11eb-8a87-b73a11919890.png)

**Comment by jnsebgosselin** [comment]
@ccordoba12

Ok, so in order to reduce the number of variables, I just did a fresh install of Anaconda and installed Spyder dependencies in a fresh new Python 3.8 environment (instead of 3.7) with `conda install -c spyder-ide/label/dev --file requirements\conda.txt`.

**Everything now looks fine**, so I think we can safely close this issue, since the problem was most likely due to my environment which was kind of outdated.

**Comment by jnsebgosselin** [comment]
I am really happy because in the code, everything looked fine, so it was puzzling me big time why this was happening XD

**Comment by ccordoba12** [comment]
Great news @jnsebgosselin! Closing then.

I also think it was a problem on your end because no one else has reported it after we released 5.0
